allennlp/training/gradient_descent_trainer.py

```diff
@@ -811,7 +812,7 @@ def _try_train(self) -> Tuple[Dict[str, Any], int]:
                     metrics["peak_" + key] = max(metrics.get("peak_" + key, 0), value)
 
             this_epoch_val_metric: float = 0.0
```

Comment:
I'm a little concerned about leaving `this_epoch_val_metric` set at 0.0 when validation is skipped. But I think we could make it `None` by default. `LearningRateScheduler.step()` and `MomentumScheduler.step()` (both called below with `this_epoch_val_metric`) take an `Optional[float]`, so that should be okay there.

```suggestion
            this_epoch_val_metric: Optional[float] = None
```

We should also be careful where `metric_track.is_best_so_far()` is checked. Maybe we only check `.is_best_so_far` when `should_validate_this_epoch` is `True`.

Reply:
What if `Model.is_best_so_far()` returned `False` if `_should_validate_this_epoch` is `False`?

Reply:
That would work too I guess, but how would `MetricTracker` know? Would the trainer set `metric_tracker._is_best_so_far` to `False` when `_should_validate_this_epoch` is `False`? It just seems cleaner for the trainer to avoid checking the metric tracker when `_should_validate_this_epoch` is `False`.

Reply:
Okay, I changed the trainer so that `should_validate_this_epoch` is checked everywhere `is_best_so_far()` is called (I count three if statements). Is that what you had in mind?

Reply:
Right, yeup!